aten/src/ATen/native/cuda/MaxUnpooling.cu

```diff
@@ -0,0 +1,591 @@
+      input.sizes());
+  AT_CHECK(
+      output_size.size() == 3,
+      "There should be exactly three elements (depth, width, height) in output_size");
+  AT_CHECK(
+      stride.size() == 3,
+      "There should be exactly three elements (depth, width, height) in stride");
+  AT_CHECK(
+      padding.size() == 3,
+      "There should be exactly three elements (depth, width, height) in padding");
+  AT_CHECK(
+      input.sizes() == indices.sizes(),
+      "Shape of indices should match shape of input");
+
+  AT_CHECK(input.numel() > 0, "Input must be non-empty");
+
+  AT_CHECK(
+      stride[0] > 0 && stride[1] > 0 && stride[2] > 0,
+      "strides should be greater than zero, but got stride: ",
+      stride);
+
+  int dimw = 3;
+  int dimh = 2;
+  int dimt = 1;
+  int dimn = 0;
+
+  if (input.ndimension() == 5) {
+    dimw++;
+    dimh++;
+    dimt++;
+    dimn++;
+  }
+
+  int nslices = input.size(dimn);
+
+  if (gradOutput.defined()) {
+    if (oT != gradOutput.size(dimt) || oH != gradOutput.size(dimh) ||
+        oW != gradOutput.size(dimw)) {
+      AT_ERROR(
+          "Inconsistent gradOutput size. oT= ",
+          oT,
+          ", oH= ",
+          oH,
+          ", oW= ",
+          oW,
+          ". gradOutput: ",
+          gradOutput.size(dimt),
+          "x",
+          gradOutput.size(dimh),
+          "x",
+          gradOutput.size(dimw));
+    }
+    AT_CHECK(
+        gradOutput.ndimension() == input.ndimension() &&
+            gradOutput.size(dimn) == nslices,
+        "gradOutput and input Tensors should have same number of dimensions and also the same number of channels/slices");
+  }
+}
+
+Tensor& max_unpooling3d_forward_out_cuda(
+    Tensor& output_,
+    const Tensor& self_,
+    const Tensor& indices_,
+    IntList output_size,
+    IntList stride,
+    IntList padding) {
+  max_unpooling3d_shape_check(
+      self_, Tensor(), indices_, output_size, stride, padding);
+
+  int64_t oT = output_size[0];
+  int64_t oW = output_size[1];
+  int64_t oH = output_size[2];
+
+  TensorArg output_arg{output_, "output_", 1}, self_arg{self_, "self_", 2},
+      indices_arg{indices_, "indices_", 3};
+  checkAllSameGPU(
+      "max_unpooling3d_forward_out_cuda", {output_arg, self_arg, indices_arg});
+
+  auto self = self_.contiguous();
+  auto indices = indices_.contiguous();
+
+  int64_t batchSize;
+  int64_t inputSlices;
+  int64_t inputTime;
+  int64_t inputHeight;
+  int64_t inputWidth;
+
+  if (self.ndimension() == 4) {
+    batchSize = 1;
+    inputSlices = self.size(0);
+    inputTime = self.size(1);
+    inputHeight = self.size(2);
+    inputWidth = self.size(3);
+    output_.resize_({inputSlices, oT, oH, oW});
+  } else {
+    batchSize = self.size(0);
+    inputSlices = self.size(1);
+    inputTime = self.size(2);
+    inputHeight = self.size(3);
+    inputWidth = self.size(4);
+    output_.resize_({batchSize, inputSlices, oT, oH, oW});
+  }
+
+  auto output = output_.contiguous(); // TODO: figure out if contiguous()
```

Comment:
Only the 3d/CUDA/forward permutation  applies `contiguous()` to its function output in the original code, so I'm replicating the inconsistency.

Reply:
Sure. There's a decent chance the original implementation was wrong, but I'm fine with being bug-for-bug compatible on this PR as long as we take note.